Add support of storage pool on boot disk.

sizzle0121 opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->
Add `storage_pool` to the boot_disk of `google_compute_instance` resource to support creating boot disk with the storage pool specified.
Since the storage pool is not a resource in the terraform provider (due to some security concern, please reference https://github.com/GoogleCloudPlatform/magic-modules/pull/10793), it's only a URL string.

<!--
Please self-review your PR against the review checklist before creating it: https://googlecloudplatform.github.io/magic-modules/contribute/review-pr/

Completing the checklist will help speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.

If your PR is still work in progress, please create it in draft mode
-->

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none

Unless you choose release-note:none, please add a release note.

See https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/ for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: added `storage_pool` under `boot_disk.initialize_params` to `google_compute_instance` resource
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-11230-old..auto-pr-11230) ( 4 files changed, 83 insertions(+), 4 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-11230-old..auto-pr-11230) ( 4 files changed, 83 insertions(+), 4 deletions(-))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_compute_instance` (288 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_instance" "primary" {
  boot_disk {
    initialize_params {
      labels = # value needed
    }
  }
}


```
Resource: `google_compute_instance_from_machine_image` (7 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_instance_from_machine_image" "primary" {
  boot_disk {
    initialize_params {
      storage_pool = # value needed
    }
  }
}


```
Resource: `google_compute_instance_from_template` (13 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_instance_from_template" "primary" {
  boot_disk {
    initialize_params {
      enable_confidential_compute = # value needed
      image                       = # value needed
      labels                      = # value needed
      provisioned_iops            = # value needed
      provisioned_throughput      = # value needed
      resource_manager_tags       = # value needed
      size                        = # value needed
      storage_pool                = # value needed
      type                        = # value needed
    }
  }
}


```

modular-magician commented:
#### Tests analytics
Total tests: 974
Passed tests: 897
Skipped tests: 74
Affected tests: 3

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>compute</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 3 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccComputeDisk_storagePoolSpecified</li>
<li>TestAccComputeInstance_bootDisk_storagePoolSpecified</li>
<li>TestAccComputeRegionPerInstanceConfig_removeInstanceOnDestroy</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$
`TestAccComputeDisk_storagePoolSpecified`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11230/artifacts/e6c27afb-947a-424a-aab6-ae24712c328e/recording/TestAccComputeDisk_storagePoolSpecified.log)]
`TestAccComputeRegionPerInstanceConfig_removeInstanceOnDestroy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11230/artifacts/e6c27afb-947a-424a-aab6-ae24712c328e/recording/TestAccComputeRegionPerInstanceConfig_removeInstanceOnDestroy.log)]

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$

---


$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccComputeInstance_bootDisk_storagePoolSpecified`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11230/artifacts/e6c27afb-947a-424a-aab6-ae24712c328e/build-log/recording_build/TestAccComputeInstance_bootDisk_storagePoolSpecified_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11230/artifacts/e6c27afb-947a-424a-aab6-ae24712c328e/recording/TestAccComputeInstance_bootDisk_storagePoolSpecified.log)]


$\textcolor{red}{\textsf{Errors occurred during RECORDING mode. Please fix them to complete your PR.}}$


View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11230/artifacts/e6c27afb-947a-424a-aab6-ae24712c328e/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-11230/artifacts/e6c27afb-947a-424a-aab6-ae24712c328e/recording) for each test

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-11230-old..auto-pr-11230) ( 3 files changed, 119 insertions(+))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-11230-old..auto-pr-11230) ( 3 files changed, 119 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_compute_instance` (288 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_instance" "primary" {
  boot_disk {
    initialize_params {
      labels = # value needed
    }
  }
}


```
Resource: `google_compute_instance_from_machine_image` (7 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_instance_from_machine_image" "primary" {
  boot_disk {
    initialize_params {
      storage_pool = # value needed
    }
  }
}


```
Resource: `google_compute_instance_from_template` (13 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_compute_instance_from_template" "primary" {
  boot_disk {
    initialize_params {
      enable_confidential_compute = # value needed
      image                       = # value needed
      labels                      = # value needed
      provisioned_iops            = # value needed
      provisioned_throughput      = # value needed
      resource_manager_tags       = # value needed
      size                        = # value needed
      storage_pool                = # value needed
      type                        = # value needed
    }
  }
}


```

modular-magician commented:
#### Tests analytics
Total tests: 974
Passed tests: 899
Skipped tests: 74
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>compute</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccComputeInstance_bootDisk_storagePoolSpecified</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{green}{\textsf{Tests passed during RECORDING mode:}}$
`TestAccComputeInstance_bootDisk_storagePoolSpecified`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11230/artifacts/2be8fd1a-4f05-430e-92ae-bac23ef85fbe/recording/TestAccComputeInstance_bootDisk_storagePoolSpecified.log)]

$\textcolor{green}{\textsf{No issues found for passed tests after REPLAYING rerun.}}$

---




$\textcolor{green}{\textsf{All tests passed!}}$

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11230/artifacts/2be8fd1a-4f05-430e-92ae-bac23ef85fbe/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-11230/artifacts/2be8fd1a-4f05-430e-92ae-bac23ef85fbe/recording) for each test

sizzle0121 commented:
Hi @trodge, we already have different test cases for the fields it mentioned, and the one I added I also added a test for it and made sure it passes, please review, thanks!
